src/navigation/restaurant/components/OrderButtons.js

```diff
@@ -25,7 +25,8 @@ const Comp = ({ order, isPrinterConnected, onPrinterClick, printOrder }) => {
         {isPrinterConnected && (
           <Button
             endIcon={<Icon as={FontAwesome} name="print" size="sm" />}
-            onPress={printOrder}>
+            onPress={printOrder}
+            isDisabled={disablePrintButton}>
```

Comment:
To be honest, I don't see much value in passing these props through three components (Order.js > OrderHeading.js > OrderButtons.js). I would put them in `OrderButtons` directly